types/jsreport-core/index.d.ts

```diff
@@ -65,50 +84,192 @@ declare namespace JsReport {
     }
 
+    type ReporterOptionsStoreProvider = 'memory';
+    type ReporterOptionsBlobStorageProvider = 'memory' | 'fs';
+    type EngineStrategy = 'dedicated-process' | 'http-server' | 'in-process';
+
     interface Configuration {
-        autoTempCleanup: boolean;
-        dataDirectory: string;
-        extensionsLocationCache: boolean;
-        loadConfig: boolean;
-        logger: {
-            silent: boolean;
+        mode?: any;
+        /** specifies where is the application root and where jsreport searches for extensions */
+        rootDirectory?: string;
+        /** specifies directory of the script that was used to start node.js, this value is mostly metadata that is useful for your own code inside jsreport scripts */
+        appDirectory?: string;
+        /**
+         * specifies where jsreport stores temporary files used by the conversion pipelineDef
+         * @default getDefaultTempDirectory()
+         */
+        tempDirectory?: string;
+        /**
+         * specifies if jsreport should load configuration values from external sources (cli args, env vars, configuration files) or not
+         * @default getDefaultLoadConfig()
+         */
+        loadConfig?: boolean;
+        /** specifies if after some interval jsreport should automatically clean up temporary files generated while rendering reports */
+        autoTempCleanup?: boolean;
+        /** specifies if jsreport should discover/search installed extensions in project and use them automatically */
+        discover?: boolean;
+        /**
+         * wheter if jsreport should read list of extensions from a previous generated cache or if it should crawl and try to search extensions again, set it to false when you want to always
+         * force crawling node_modules when searching for extensions while starting jsreport
+         * @default true
+         */
+        useExtensionsLocationCache?: boolean;
+        logger?: {
+            silent?: boolean;
+        };
+        /** global single timeout that controls how much a report generation should wait before it times out */
+        reportTimeout?: number;
+        /**
+         * option that enables passing a custom report timeout per request using req.options.timeout. this enables that the caller of the report generation control the report timeout so enable it
+         * only when you trust the caller
+         * @default false
+         */
+        enableRequestReportTimeout?: boolean;
+        /** @default false */
+        allowLocalFilesAccess?: boolean;
+        encryption?: {
+            /** lenght must be 16 characters */
+            secretKey: string;
+            /** @default true */
+            enabled?: boolean;
+        };
+        templatingEngines?: {
+            /** @default 'dedicated-process' */
+            strategy?: EngineStrategy;
+            numberOfWorkers?: number;
+            forkOptions?: {
+                execArgv?: string | [string];
```

Comment:
is `[string]` is on purpose ? or it should be `string[]`?

Reply:
I've been racking my head on where I read some recommendations for using [T] vs T[] in this capacity but I can't find it at the moment.

Reply:
It seems not like that.
[string] means an array contains only one string element. cannot be 0 or more than 1.